### ISSUE TITLE
FIO benchmark with small files shows higher bandwidth than network bandwidth. 

### ISSUE THREAD

**Issue by Yeosangho** [seaweedfs/seaweedfs#5157]
**Describe the bug**
FIO benchmark with small files shows higher bandwidth than network bandwidth. 
1. When file is smaller than chunkSize in mount, file read bandwidth is higher than network bandwidth.
2. When file is small size(1M~100M), file write bandwidth is higher than network bandwidth. 

**System Setup**
- List the command line to start "weed master", "weed volume", "weed filer", "weed s3", "weed mount".

weed master -mdir=/mnt/metadata -peers=10.10.30.25:9334 -ip=10.10.30.25 -port=9334 -volumeSizeLimitMB=512 > seaweed.log 2>&1 &

weed volume -mserver=10.10.30.25:9334 -ip=10.10.30.25 -port=8080 -dir=/mnt/weed/data -dir.idx=/mnt/weed/metadata -idleTimeout=60 -index=memory -max=0 -readMode=proxy -hasSlowRead=true -readBufferSizeMB=64 -concurrentDownloadLimitMB=5 -concurrentUploadLimitMB=5 -rack=A -dataCenter=dc1 > seaweed.log 2>&1 &

weed filer -master=10.10.30.25:9334-ip=10.10.30.25 -port=8888 -ui.deleteDir=false -maxMB=4 -rack A > seaweed.log 2>&1 &

weed mount -filer=10.10.30.25:8888 -dir=/mnt/node22-seaweed-3 -filer.path=/ -cacheDir=/home/surromind/shyeo/seaweedfs-cache -cacheCapacityMB=0 

- OS version
Seaweed Server : Ubuntu 22.04.1 LTS
Seaweed FUSE Client :  Ubuntu 20.04.6 LTS

- output of `weed version` : version 30GB 3.57 0f8168c0c928bba3d2f48b0680d3bdce9c617559 linux amd64

- if using filer, show the content of `filer.toml`
`[leveldb2]
enabled = true
dir = "/data/filerldb2"`

**Expected behavior**
the file I/O bandwidth of fio benchmark result should be under the network bandwidth(1000Mbit/s)

**Screenshots**
1. [weed benchmark] when I run "weed benchmark -master=10.10.30.25:9334 -size=1000000 -fsync=true", It shows lower bandwidth than network bandwidth. 
 
![image](https://github.com/seaweedfs/seaweedfs/assets/21786047/0d02ef0e-c96c-4e97-a6a9-19f96060803c)

2. [File Write] when I run "fio --directory=/mnt/node22-seaweed-3/10M --name=fio_test37  --nrfiles=1  --direct=1   --rw=randwrite  --bs=4K     --size=10M  --numjobs=100 --iodepth=8  --time_based --runtime=20", It shows higher network bandwidth than network bandwidth
![image](https://github.com/seaweedfs/seaweedfs/assets/21786047/7d006c05-7521-4f82-84df-e0a8c1ee51f5)

3. [File Read] when I run "fio --directory=/mnt/node22-seaweed-3/10M --name=fio_test37  --nrfiles=1  --direct=1   --rw=randread  --bs=4K     --size=10M  --numjobs=100 --iodepth=8  --time_based --runtime=20", It shows higher network bandwidth than network bandwidth

![image](https://github.com/seaweedfs/seaweedfs/assets/21786047/31cb71f8-0fb3-4a0f-b452-ba19e18f2bbf)

**Comment by Yeosangho** [comment]
I think seaweed + fio causes some unexpected problem, Thus I made script to check file io performance with dd command. 
```
#!/bin/bash

number_of_trials=4        # Set the number of trials to run
number_of_files=1       # Set the number of files you want to create/read
file_size="1M"            # Set the size of each file (e.g., 1G for 1 Gigabyte)
output_dir="./10M"  # Set the directory where files will be created/read
output_dir="/mnt/node22-seaweed-3/10M"
total_write_time=0
total_read_time=0

for trial in $(seq 1 $number_of_trials); do
    echo "Starting trial $trial"

    # Write phase
    write_start_time=$(date +%s.%N)  # Start time for write phase
    for i in $(seq 1 $number_of_files); do
        output_file="${output_dir}/file_${trial}_${i}.dat"
        echo "Creating file $i for trial $trial"
        dd if=/dev/urandom of="$output_file" oflag=direct bs=4K count=256 conv=fdatasync &
    done
    wait  # Wait for all write jobs to finish
    write_end_time=$(date +%s.%N)  # End time for write phase
    write_time=$(echo "$write_end_time - $write_start_time" | bc)
    total_write_time=$(echo "$total_write_time + $write_time" | bc)
    echo "Write time for trial $trial: $write_time seconds"

    # Read phase
    read_start_time=$(date +%s.%N)  # Start time for read phase
    for i in $(seq 1 $number_of_files); do
        input_file="${output_dir}/file_${trial}_${i}.dat"
        dd if="$input_file" of=/dev/null iflag=direct bs=4K &
    done
    wait  # Wait for all read jobs to finish
    read_end_time=$(date +%s.%N)  # End time for read phase
    read_time=$(echo "$read_end_time - $read_start_time" | bc)
    total_read_time=$(echo "$total_read_time + $read_time" | bc)
    echo "Read time for trial $trial: $read_time seconds"
done

# Compute the average times
average_write_time=$(echo "$total_write_time / $number_of_trials" | bc -l)
average_read_time=$(echo "$total_read_time / $number_of_trials" | bc -l)

echo "Average write time per trial: $average_write_time seconds"
echo "Average read time per trial: $average_read_time seconds"


```

**Comment by Yeosangho** [comment]
I thought my problem was fixed by dd scripts, However it is unresolved. 
1. DD command can not benchmark the random RW. 
2. when I make the file with /dev/zero, not /dev/urandom, the bandwidth of read operation is over the its own network bandwidth again..

**Comment by Yeosangho** [comment]
I finally found the solution to get the proper RW bandwidth(under the network bandwidth) by fio. 
I removed `--time_based --runtime={seconds}` and increase --size to 1G and split files by --nrfiles. 
I assumed that there is some memory cache features in weed mount, but I couldn't expect the cahce features. 

As a result, the fio benchmark command as follows

1. File Read 1M
```
fio --directory=/mnt/node22-seaweed-3/10M --name=fio_test44  --nrfiles=1000 --direct=1   --rw=read  --size=1G  --numjobs=8 --bs=4K
```


2. File Write 1M
```
fio --directory=/mnt/node22-seaweed-3/10M --name=fio_test44  --nrfiles=1000 --direct=1   --rw=write --size=1G  --numjobs=8 --bs=4K
```

Both command result in 100MB/s bandwidth which is lower than network bandwidth 125MB/s
Furthermore, when I decrease the number in --numjobs in command as 1, the RW bandwidth is also decreased. This decreased bandwidth also checks in benchmark tools in seaweed.
